ic-types/src/principal.rs

```diff
@@ -136,8 +161,35 @@ impl Principal {
     }
 
     /// An anonymous Principal.
-    pub fn anonymous() -> Self {
-        Self(From::<&[u8]>::from(&[PrincipalClass::Anonymous as u8]))
+    pub const fn anonymous() -> Self {
+        Self(PrincipalInner::from_slice(&[
+            PrincipalClass::Anonymous as u8
+        ]))
+    }
+
+    /// Attempt to decode a slice into a Principal. `panic`s if the bytes can't be interpreted.
+    pub const fn from_slice(bytes: &[u8]) -> Self {
+        if let Ok(v) = Self::try_from_slice(bytes) {
+            v
+        } else {
+            // TODO: replace with panic (blocked by rust-lang/rust#85194)
+            const_panic!("slice length exceeded capacity");
```

Comment:
Does this translate into a regular `panic!()` when the function is called as a non-const?